impl/src/main/java/org/jboss/weld/context/http/HttpRequestContextImpl.java

```diff
@@ -49,11 +51,13 @@ public HttpRequestContextImpl(String contextId) {
 
     public boolean associate(HttpServletRequest request) {
         // At this point the bean store should never be set - see also HttpContextLifecycle#nestedInvocationGuard
-        if (getBeanStore() != null) {
+        BoundBeanStore beanStore = getBeanStore();
+        if (beanStore != null) {
             ContextLogger.LOG.beanStoreLeakDuringAssociation(this.getClass().getName(), request);
+            ContextLogger.LOG.beanStoreLeakAffectedBeanIdentifiers(this.getClass().getName(), Iterables.toMultiRowString(beanStore));
```

Comment:
We should only be building the multi-row string if leak happens AND debug logging is enabled.